pkgs/build-support/rust/replace-workspace-values.py

```diff
@@ -19,53 +19,51 @@ def load_file(path: str) -> dict[str, Any]:
 # See `inner_dependency_inherit_with`:
 # https://github.com/rust-lang/cargo/blob/4de0094ac78743d2c8ff682489e35c8a7cafe8e4/src/cargo/util/toml/mod.rs#L982
 def replace_key(
-    workspace_manifest: dict[str, Any], table: dict[str, Any], section: str, key: str
+    workspace_manifest: dict[str, Any],
+    table: dict[str, Any],
+    section: Literal["package", "dependencies"],
+    key: str,
 ) -> bool:
-    if (
-        isinstance(table[key], dict)
-        and "workspace" in table[key]
-        and table[key]["workspace"] is True
-    ):
-        print("replacing " + key)
+    if not isinstance(table[key], dict) or table[key].get("workspace") is not True:
+        return False
+    print("replacing " + key)
 
-        local_dep = table[key]
-        del local_dep["workspace"]
+    local_dep = table[key]
+    del local_dep["workspace"]
 
-        workspace_dep = workspace_manifest[section][key]
+    workspace_dep: str | dict[str, Any] = workspace_manifest[section][key]
 
-        if section == "dependencies":
-            if isinstance(workspace_dep, str):
-                workspace_dep = {"version": workspace_dep}
+    if section == "package":
+        table[key] = workspace_dep
+        return True
 
-            final: dict[str, Any] = workspace_dep.copy()
+    _ = assert_type(section, Literal["dependencies"])
```

Comment:
This would probably be better with an explicit `assert` or `if section != "dependencies": raise ...`

Reply:
You are probably right, I might've forgotten that there's no type checking ran be nixpkgs tests here, if that's the case.

Reply:
UPD: no, it is not needed. `section` can only be `"dependencies"` or `"package"`, which is what it's called with, there's no need to check anything in runtime.